### ISSUE TITLE
Proposals: CEXL  / CEL improvements

### ISSUE THREAD

**Issue by mikesimons** [istio/istio#12454]
**Describe the feature request**
We tried to create an additional dimension for response.code to bucket it by family (e.g. 1xx, 2xx, 3xx etc) as per https://discuss.istio.io/t/expanding-cexl/1355 but found it impossible to do in 1.0.X. It should be possible in 1.1 but is less than ideal. We aren't running 1.1 yet but I suspect this will work:

```
condition(response.code >= 100 && response.code < 200, "1XX", condition(response.code >= 200 && response.code < 300, "2XX", condition(response.code >= 300 && response.code < 400, "3XX", condition(response.code >= 400 && response.code < 500, "4XX, condition(response.code >= 500 && response.code < 600, "5XX", "unknown)))))
```

This seems overly verbose and could be shorted with some fairly easy to implement features (I think).

If the third argument to `conditional` were optional (and default to an empty value) then we could use the pipe operator rather than nested calls:

If condition could make the 3rd argument optional such that it could be used with `|`

**Example**
`conditional(response.code >= 200 && response.code < 300, "2XX") | conditional(response.code >= 300 && response.code < 400, "3XX")`

Shortening / aliasing `conditional` to `if` would also make it a bit more familiar (and terse):

**Example**
`if(response.code >= 200 && response.code < 300, "2XX")`

Ranges would also make similar use cases nicer (although they'd require a new data type to be efficient). 

**Example**
`if(response.code in range(200, 299), "2XX") | if(response.code in range(300, 399), "3XX")`

**Comment by dsanchor** [comment]
I was trying to set the requestcount metric label 'response_code'  to:
` conditional((api.protocol | context.protocol | "unknown") == "grpc", response.grpc_status, (response.code | 200) )`
so I could have both gRPC and HTTP response codes in the same label. 

The problem is that response.grpc_status is an String and response.code is int64, so mixer validation fails due to [type checking](https://istio.io/docs/reference/config/policy-and-telemetry/expression-language/#type-checking).

It would be good to have kind of a converter function or so to avoid these kind of issues. Example:
` conditional((api.protocol | context.protocol | "unknown") == "grpc", strconv.ParseInt(response.grpc_status, 10, 64), (response.code | 200))`

**Comment by kyessenov** [comment]
@dsanchor thanks for the feedback! Both `int(_)` and `string(_)` are obvious candidates to help with response codes. I'm not sure why the gRPC status is even a string TBH, an enum is more appropriate.

**Comment by kyessenov** [comment]
Ability to parse URL to split path/host was requested here https://discuss.istio.io/t/how-to-split-value-in-telemetry/3107

**Comment by BenPope** [comment]
As above, it would be really useful to be able to do regex replace, e.g., extract portions of a URL into new atttributes, or even reorder path segments.

**Comment by josephpeacock** [comment]
Hello. Reopening as I've run into a situation where my team needs the ability to replace parts of an istio attribute for our logging. More specifically, we'd like to only use everything after a slash, for example `foobar/whatever` should be logged as `whatever`, but a regex-based replace seems like a more generic way to solve it.

I see that @JacksChoco had a PR for this that was closed without a merge. I've recreated the PR from my own fork (https://github.com/istio/istio/pull/22110) in the hopes of getting this included.
